### PR TITLE
Feature/optional file metadata

### PR DESCRIPTION
**Pull request by mgdaily**
This PR adds the ability to upload files of arbitrary type to an OCS Science Archive. This is accomplished by specifying a dictionary of "file metadata" when ingesting a file. This dictionary is treated exactly the same as a FITS header and is validated against the `required_headers` settings.

* Add helper method for File class to use Astropy to check if it's a valid FITS file. If not, expect that we've been provided some metadata and use that to construct a FitsDict
* Add test PDF/add tests for PDF uploads
* Re-organize test data a bit (46 changed files 😱) - `tests/fits/` has now moved to `tests/test_files/fits` and there is now a `tests/test_files/other` directory where we can throw any other weird filetypes we want. Some FITS-specific tests iterate through all of the test FITS files, so having them separate is nice.
* Update some docstrings to reflect this new functionality.
* Add new header keywords L1IDARC, L1ID1D, L1ID2D, L1IDSUM for BANZAI-NRES' related frames

### CODE REVIEW

**Review by jnation3406 — APPROVED** [review]
All looks good to me

File: ocs_ingester/fits.py [hunk 1]
```diff
@@ -34,23 +34,49 @@ def __init__(self, file, required_headers, blacklist_headers):
...
+        """
+        Check some file metadata for required headers.
+
+        :param metadata_dict: dictionary of file metadata
+        :return True if required headers are present, False if not
+        """
+        if any([k for k in self.required_headers if k not in metadata_dict]):
+            return False
+        else:
+            logger.info('Ingester extracted fits headers', extra={
```
Comment (eheinrich) [inline ocs_ingester/fits.py:72]: I think this message could be misleading when it's being printed out when validating file metadata that was not extracted from a FITS file.... I mean, I guess it is technically still information that was in the associated FITS file header, but it wasn't extracted by the ingester.

File: ocs_ingester/utils/fits.py [hunk 2]
```diff
@@ -29,6 +32,19 @@ def get_from_start(self):
...
         return self.fileobj
 
+    def is_valid_fits(self):
+        """
+        Use astropy to determine if a file is a valid FITS file
+        :return: True if valid FITS, False if not
+        """
+        try:
+            with self.get_fits() as fits_file:
+                hdu_list = fits.open(io.BytesIO(fits_file.read()), mode='readonly')
```
Comment (eheinrich) [inline ocs_ingester/utils/fits.py:42]: Do you have to call a close method on hdu_list?

**Review by eheinrich — COMMENTED** [review]
Looks good, just a few questions.

File: ocs_ingester/fits.py (hunk 1 above)
Comment (mgdaily) [reply ocs_ingester/fits.py:72]: Good call. I think we can change this to "Ingester validated/extracted file metadata" or something like that, so it's more generalized.

File: ocs_ingester/scripts/ingest_frame.py [hunk 3]
```diff
@@ -17,8 +17,8 @@
 from ocs_ingester.exceptions import NonFatalDoNotRetryError
 
 description = (
-    'Upload a file to the science archive of an observatory control system. This script will output the resulting URL '
-    'if the upload is successful. An optional flag --check-only can be used to check for the existence of a file '
+    'Upload a FITS file to the science archive of an observatory control system. This script will output the resulting '
```
[the remark this answers is not in the document]
Comment (mgdaily) [reply ocs_ingester/scripts/ingest_frame.py:20]: Yep, exactly. If we want to add the ability to parse a JSON file I suppose we can. But for now I just assumed it'd be used for FITS files.

File: ocs_ingester/utils/fits.py (hunk 2 above)
Comment (mgdaily) [reply ocs_ingester/utils/fits.py:42]: I am not sure - will investigate. It'd probably be safe to. I use the `get_fits` context manager which closes the file object, but I'll check on the hdu_list.

Comment (mgdaily) [reply ocs_ingester/utils/fits.py]: I went ahead and replicated what was done [here]( https://github.com/observatorycontrolsystem/ingester/blob/9281b5552474ecd09d163e4e3fbc607578f86003/ocs_ingester/fits.py#L40), by using `fits.open` as a context manager. I think it's a safer bet. Thanks for finding that!

File: ocs_ingester/fits.py (hunk 1 above)
Comment (mgdaily) [reply ocs_ingester/fits.py]: Updated to say: "Ingester validated file metadata," which is more in-line with what the code is _actually_ doing.

**Review by eheinrich — APPROVED** [review]

**Pull request merged**